ci-operator/step-registry/baremetal/lab/agent/day2/baremetal-lab-agent-day2-commands.sh

```diff
@@ -0,0 +1,309 @@
+  scp "${SSHOPTS[@]}" "${DAY2_INSTALL_DIR}"/boot-artifacts/agent.*-initrd* \
+   "root@${AUX_HOST}:/opt/dnsmasq/tftpboot/${CLUSTER_NAME}/initramfs_${gnu_arch}.img"
+  scp "${SSHOPTS[@]}" "${DAY2_INSTALL_DIR}"/boot-artifacts/agent.*-rootfs* \
+   "root@${AUX_HOST}:/opt/html/${CLUSTER_NAME}/rootfs-${gnu_arch}.img"
+;;
+*)
+  echo "Unknown install mode: ${BOOT_MODE}"
+  exit 1
+esac
+
+day2_IPs=""
+
+# shellcheck disable=SC2154
+for bmhost in $(yq e -o=j -I=0 '.[] | select(.name|test("-a-"))' "${SHARED_DIR}/hosts.yaml"); do
+  # shellcheck disable=SC1090
+  . <(echo "$bmhost" | yq e 'to_entries | .[] | (.key + "=\"" + .value + "\"")')
+  day2_IPs+="${ip},"
+  echo "Power on #${host} (${name})..."
+  timeout -s 9 10m ssh "${SSHOPTS[@]}" "root@${AUX_HOST}" prepare_host_for_boot "${host}" "${BOOT_MODE}"
+done
+
+day2_IPs=${day2_IPs%,}
+day2_workers=0
+touch /tmp/output.txt
+
+echo "Monitoring day2 workers and pending CSRs..."
+
+/cli/oc adm node-image monitor --ip-addresses "${day2_IPs}" -a "${day2_pull_secret}" --insecure=true 2>&1 | \
+  tee /tmp/output.txt | while IFS= read -r line; do
+  echo "$line"
+  if [[ "$line" = *"with signerName kubernetes.io/kube-apiserver-client-kubelet and username system:serviceaccount:openshift-machine-config-operator:node-bootstrapper is Pending and awaiting approval"* ]]; then
+    node_ip=$(echo "$line" | sed 's/^.*Node \(.*\): CSR.*$/\1/')
+    csr=$(echo "$line" | sed 's/^.*CSR \([^ ]*\).*$/\1/')
+    echo "Approving CSR $csr for node $node_ip"
+    oc adm certificate approve "$csr"
+  fi
+  if [[ "$line" = *"with signerName kubernetes.io/kubelet-serving and username "*" is Pending and awaiting approval"* ]]; then
+    node_ip=$(echo "$line" | sed 's/^.*Node \(.*\): CSR.*$/\1/')
+    csr=$(echo "$line" | sed 's/^.*CSR \([^ ]*\).*$/\1/')
+    if oc get nodes -o wide | grep "$node_ip" | awk '$2 != "NotReady" {print $1,$2}'; then
+       echo "$node_ip status should be NotReady, something wrong with Monitoring..."
+    fi
+    echo "Approving CSR $csr for node $node_ip"
+    oc adm certificate approve "$csr"
+  fi
+  if [[ "$line" = *"Node is Ready"* ]]; then
+    ((day2_workers+=1))
+    if [[ "$day2_workers" == "$ADDITIONAL_WORKERS" ]]; then
+       if [[ "$(oc get nodes | grep "\-""a""\-" | grep -c Ready)" == "$day2_workers" ]]; then
+         echo "All day2 workers are ready, install success..."
+       else
+         echo "Not all day2 worker is ready, something wrong with Monitoring, install fail..."
+         exit 1
+       fi
+    else
+       echo "Not all day2 worker is ready, please wait..."
+    fi
+  fi
+done
+
+EXIT_STATUS="${PIPESTATUS[0]}"
+rm -f "${day2_pull_secret}"
+if [[ $EXIT_STATUS != 0 ]]; then
+  echo "Exiting with status $EXIT_STATUS"
+  exit $EXIT_STATUS
+fi
+
+# Add operators status checking until monitoring enhanced to do this
+echo "Check all cluster operators get stable and ready"
+wait_clusteroperators_continous_success
```

Comment:
Both functions `check_clusteroperators` and `wait_clusteroperators_continuous_success` can be removed. The suggestion below will check for all three conditions: `Available=True`, `Degraded=False`, and `Progressing=False`.
```suggestion
oc adm wait-for-stable-cluster --minimum-stable-period=3m --timeout=15m
```